internal/service/remotecfg/config_manager.go

```diff
@@ -0,0 +1,373 @@
+func (cm *configManager) fetchLoadConfig(ctx fetchContext, allowCacheFallback bool) {
+	if err := cm.fetchLoadRemoteConfig(ctx); err != nil && err != errNotModified {
+		if allowCacheFallback {
+			cm.logger.Log("level", "error", "msg", "failed to fetch remote config, falling back to cache", "err", err)
+			cm.fetchLoadLocalConfig()
+		} else {
+			cm.logger.Log("level", "debug", "msg", "failed to fetch remote config, continuing with current config", "err", err)
+		}
+	}
+}
+
+func (cm *configManager) fetchLoadRemoteConfig(ctx fetchContext) error {
+	cm.logger.Log("level", "debug", "msg", "fetching remote configuration")
+
+	gcr, err := ctx.getAPIConfig()
+	cm.metrics.totalAttempts.Add(1)
+
+	// Handle "not modified" response specifically
+	if err == errNotModified {
+		cm.logger.Log("level", "debug", "msg", "skipping over API response since it has not been modified since last fetch")
+		cm.metrics.lastFetchNotModified.Set(1)
+		return nil
+	}
+
+	// Handle other errors
+	if err != nil {
+		cm.logger.Log("level", "error", "msg", "failed to fetch remote config", "err", err)
+		cm.metrics.totalFailures.Add(1)
+		cm.metrics.lastLoadSuccess.Set(0)
+		return err
+	}
+
+	// Success case - we have valid config data from the API
+	cm.metrics.lastFetchSuccessTime.SetToCurrentTime()
+	cm.metrics.lastFetchNotModified.Set(0)
+
+	// Store the remote hash from the API response
+	if gcr.Hash != "" {
+		cm.mut.Lock()
+		cm.remoteHash = gcr.Hash
+		cm.mut.Unlock()
+	}
+
+	b := []byte(gcr.GetContent())
+	newConfigHash := getHash(b)
+
+	// Check if we already received this exact config from remote
+	alreadyReceived := cm.getLastReceivedCfgHash() == newConfigHash
+	alreadyLoaded := cm.getLastLoadedCfgHash() == newConfigHash
+
+	if alreadyReceived {
+		cm.logger.Log("level", "debug", "msg", "skipping over API response since it matched the last received one",
+			"config_hash", newConfigHash, "already_loaded", alreadyLoaded)
+		return nil
+	}
+
+	// Record that we received this config from remote (regardless of parse success)
+	cm.setLastReceivedCfgHash(newConfigHash)
+
+	err = cm.parseAndLoad(b)
+	if err != nil {
+		// Failed to parse/load the configuration - received hash is recorded, but loaded hash unchanged
+		cm.logger.Log("level", "error", "msg", "failed to parse remote config",
+			"received_hash", newConfigHash, "loaded_hash", cm.getLastLoadedCfgHash(), "err", err)
+		cm.metrics.lastLoadSuccess.Set(0)
+		return err
+	}
+
+	// Successfully loaded the configuration - now update the loaded hash
+	cm.setLastLoadedCfgHash(newConfigHash)
+	cm.metrics.lastLoadSuccess.Set(1)
+
+	cm.logger.Log("level", "info", "msg", "successfully loaded remote configuration",
+		"config_hash", newConfigHash, "config_size", len(b))
+
+	// If successful, flush to disk and keep a copy.
+	cm.setCachedConfig(b)
+	return nil
+}
+
+func (cm *configManager) fetchLoadLocalConfig() {
+	cachePath := cm.getCachedConfigPath()
+	b, err := cm.getCachedConfig()
```

Comment:
Why do we need the cachePath here, is it just for the logging line? Then maybe we can move it inline with the logger just for clarity.